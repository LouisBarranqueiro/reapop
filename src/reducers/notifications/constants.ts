export enum NotificationAction {
    UpsertNotification = 'reapop/upsertNotification',
    DismissNotification = 'reapop/dismissNotification',
    DismissNotifications = 'reapop/dismissNotifications',
}
