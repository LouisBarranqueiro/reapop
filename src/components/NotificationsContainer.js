import React, {Component} from 'react';
import TransitionGroup from 'react/lib/ReactCSSTransitionGroup';
import Notification from './Notification';
import {POSITIONS} from '../constants';

export class NotificationsContainer extends Component {
  // default types
  static defaultProps = {
    notifications: []
  };

  // Properties types
  static propTypes = {
    notifications: React.PropTypes.array.isRequired,
    position: React.PropTypes.string.isRequired,
    theme: React.PropTypes.shape({
      notificationsContainer: React.PropTypes.shape({
        className: React.PropTypes.shape({
          main: React.PropTypes.string.isRequired,
          position: React.PropTypes.func.isRequired
        }).isRequired,
        transition: React.PropTypes.shape({
          name: React.PropTypes.object.isRequired,
          enterTimeout: React.PropTypes.number.isRequired,
          leaveTimeout: React.PropTypes.number.isRequired
        }).isRequired
      }).isRequired,
      notification: React.PropTypes.shape({
        className: React.PropTypes.object.isRequired
      }).isRequired
    }).isRequired
  };

  /**
   * Constructor
   * Bind methods
   * @param {Object} props
   * @returns {void}
   */
  constructor(props) {
    super(props);
    this._renderNotifications = this._renderNotifications.bind(this);
  }

  /**
   * Render notifications
   * @private
   * @returns {XML}
   */
  _renderNotifications() {
    // get all notifications and default values for notifications
    const {
      position,
      theme: {
        notification: {className}
      }
    } = this.props;
    let {notifications} = this.props;

    // when notifications are displayed at the bottom,
    // we display notifications from bottom to top
    if ([POSITIONS.bottomLeft, POSITIONS.bottomRight, POSITIONS.bottomCenter, POSITIONS.bottom]
        .indexOf(position) >= 0) {
      notifications = notifications.reverse();
    }

    return notifications.map((notification) => (
      <Notification key={notification.id} notification={notification} className={className}/>
    ));
  }

  /**
   * Render
   * @returns {XML}
   */
  render() {
    const {
      className, transition: {name, enterTimeout, leaveTimeout}
    } = this.props.theme.notificationsContainer;
    const {position} = this.props;

    return (
      <div className={`${className.main} ${className.position(position)}`}>
        <TransitionGroup
          transitionName={name}
          transitionEnterTimeout={enterTimeout}
          transitionLeaveTimeout={leaveTimeout}>
          {this._renderNotifications()}
        </TransitionGroup>
      </div>
    );
  }
}

export default NotificationsContainer;
