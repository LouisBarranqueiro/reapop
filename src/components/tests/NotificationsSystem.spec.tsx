import React from 'react'
import {act, render} from '@testing-library/react'
import pretty from 'pretty'

import {Notification} from '../..'
import {POSITIONS, STATUSES} from '../../constants'
import atalhoTheme from '../../themes/atalho'
import NotificationsSystem from '../NotificationsSystem'
import NotificationsContainer from '../NotificationsContainer'

jest.mock('../SlideTransition', () => jest.fn((props) => props.children))
jest.mock('../NotificationsContainer', () => {
    const component = jest.requireActual('../NotificationsContainer').default
    return jest.fn(component)
})

describe('<NotificationsSystem/>', () => {
    const notifications: Notification[] = Object.values(POSITIONS).map((position, index) => {
        return {
            id: index.toString(),
            position,
            message: 'hello world!',
            status: STATUSES.info,
            buttons: [],
        }
    })

    it('should display notifications in each container and then in the top container', () => {
        const windowInitialHeight = window.innerHeight
        const windowInitialWidth = window.innerWidth
        const {container} = render(
            <NotificationsSystem notifications={notifications} dismissNotification={jest.fn()} />
        )
        expect(pretty(container.innerHTML)).toMatchSnapshot()

        act(() => {
            window.resizeTo(100, 100)
        })

        expect(pretty(container.innerHTML)).toMatchSnapshot()
        expect((NotificationsContainer as jest.Mock).mock.calls).toMatchSnapshot()

        act(() => {
            // resize window back to its initial size
            window.resizeTo(windowInitialWidth, windowInitialHeight)
        })
    })

    it.each([
        ['without theme', undefined],
        ['with theme', atalhoTheme],
    ])('should display notifications %s', (_, theme) => {
        const {container} = render(
            <NotificationsSystem
                notifications={notifications.slice(0, 1)}
                dismissNotification={jest.fn()}
                theme={theme}
            />
        )
        expect(pretty(container.innerHTML)).toMatchSnapshot()
    })

    it.each([
        ['default components', {}],
        [
            'custom components (transition and icon)',
            {
                Transition: jest.fn((props) => props.children),
                NotificationIcon: function CustomIcon() {
                    return <div>custom-icon</div>
                },
            },
        ],
        [
            'custom components (transition and notification)',
            {
                Transition: jest.fn((props) => props.children),
                Notification: function CustomNotification() {
                    return <div>custom-notification</div>
                },
            },
        ],
    ])('should display notifications with %s', (_, components) => {
        const {container} = render(
            <NotificationsSystem
                notifications={notifications.slice(0, 1)}
                dismissNotification={jest.fn()}
                components={components}
            />
        )

        if (components.hasOwnProperty('Transition')) {
            expect((components as {Transition: jest.Mock}).Transition.mock.calls).toMatchSnapshot()
        }
        expect(pretty(container.innerHTML)).toMatchSnapshot()
    })
})
