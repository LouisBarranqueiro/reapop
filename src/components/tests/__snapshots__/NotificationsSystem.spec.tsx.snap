// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotificationsSystem/> should display notifications in each container and then in the top container 1`] = `
"<div>
  <div class=\\"reapop__container reapop__container--top-center\\">
    <div>
      <div data-testid=\\"timed-notification\\">
        <div data-testid=\\"notification\\" role=\\"alert\\" class=\\"reapop__notification reapop__notification--info reapop__notification--buttons-0 \\"><svg viewBox=\\"0 0 16 16\\" fill=\\"currentColor\\" class=\\"reapop__notification-dismiss-icon\\">
            <path fill-rule=\\"evenodd\\" d=\\"M8 16A8 8 0 1 0 8 0a8 8 0 0 0 0 16zm.93-9.412l-2.29.287-.082.38.45.083c.294.07.352.176.288.469l-.738 3.468c-.194.897.105 1.319.808 1.319.545 0 1.178-.252 1.465-.598l.088-.416c-.2.176-.492.246-.686.246-.275 0-.375-.193-.304-.533L8.93 6.588zM8 5.5a1 1 0 1 0 0-2 1 1 0 0 0 0 2z\\"></path>
          </svg>
          <div class=\\"reapop__notification-meta\\">
            <p class=\\"reapop__notification-message\\">hello world!</p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div class=\\"reapop__container reapop__container--top-left\\">
    <div>
      <div data-testid=\\"timed-notification\\">
        <div data-testid=\\"notification\\" role=\\"alert\\" class=\\"reapop__notification reapop__notification--info reapop__notification--buttons-0 \\"><svg viewBox=\\"0 0 16 16\\" fill=\\"currentColor\\" class=\\"reapop__notification-dismiss-icon\\">
            <path fill-rule=\\"evenodd\\" d=\\"M8 16A8 8 0 1 0 8 0a8 8 0 0 0 0 16zm.93-9.412l-2.29.287-.082.38.45.083c.294.07.352.176.288.469l-.738 3.468c-.194.897.105 1.319.808 1.319.545 0 1.178-.252 1.465-.598l.088-.416c-.2.176-.492.246-.686.246-.275 0-.375-.193-.304-.533L8.93 6.588zM8 5.5a1 1 0 1 0 0-2 1 1 0 0 0 0 2z\\"></path>
          </svg>
          <div class=\\"reapop__notification-meta\\">
            <p class=\\"reapop__notification-message\\">hello world!</p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div class=\\"reapop__container reapop__container--top-right\\">
    <div>
      <div data-testid=\\"timed-notification\\">
        <div data-testid=\\"notification\\" role=\\"alert\\" class=\\"reapop__notification reapop__notification--info reapop__notification--buttons-0 \\"><svg viewBox=\\"0 0 16 16\\" fill=\\"currentColor\\" class=\\"reapop__notification-dismiss-icon\\">
            <path fill-rule=\\"evenodd\\" d=\\"M8 16A8 8 0 1 0 8 0a8 8 0 0 0 0 16zm.93-9.412l-2.29.287-.082.38.45.083c.294.07.352.176.288.469l-.738 3.468c-.194.897.105 1.319.808 1.319.545 0 1.178-.252 1.465-.598l.088-.416c-.2.176-.492.246-.686.246-.275 0-.375-.193-.304-.533L8.93 6.588zM8 5.5a1 1 0 1 0 0-2 1 1 0 0 0 0 2z\\"></path>
          </svg>
          <div class=\\"reapop__notification-meta\\">
            <p class=\\"reapop__notification-message\\">hello world!</p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div class=\\"reapop__container reapop__container--bottom-center\\">
    <div>
      <div data-testid=\\"timed-notification\\">
        <div data-testid=\\"notification\\" role=\\"alert\\" class=\\"reapop__notification reapop__notification--info reapop__notification--buttons-0 \\"><svg viewBox=\\"0 0 16 16\\" fill=\\"currentColor\\" class=\\"reapop__notification-dismiss-icon\\">
            <path fill-rule=\\"evenodd\\" d=\\"M8 16A8 8 0 1 0 8 0a8 8 0 0 0 0 16zm.93-9.412l-2.29.287-.082.38.45.083c.294.07.352.176.288.469l-.738 3.468c-.194.897.105 1.319.808 1.319.545 0 1.178-.252 1.465-.598l.088-.416c-.2.176-.492.246-.686.246-.275 0-.375-.193-.304-.533L8.93 6.588zM8 5.5a1 1 0 1 0 0-2 1 1 0 0 0 0 2z\\"></path>
          </svg>
          <div class=\\"reapop__notification-meta\\">
            <p class=\\"reapop__notification-message\\">hello world!</p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div class=\\"reapop__container reapop__container--bottom-left\\">
    <div>
      <div data-testid=\\"timed-notification\\">
        <div data-testid=\\"notification\\" role=\\"alert\\" class=\\"reapop__notification reapop__notification--info reapop__notification--buttons-0 \\"><svg viewBox=\\"0 0 16 16\\" fill=\\"currentColor\\" class=\\"reapop__notification-dismiss-icon\\">
            <path fill-rule=\\"evenodd\\" d=\\"M8 16A8 8 0 1 0 8 0a8 8 0 0 0 0 16zm.93-9.412l-2.29.287-.082.38.45.083c.294.07.352.176.288.469l-.738 3.468c-.194.897.105 1.319.808 1.319.545 0 1.178-.252 1.465-.598l.088-.416c-.2.176-.492.246-.686.246-.275 0-.375-.193-.304-.533L8.93 6.588zM8 5.5a1 1 0 1 0 0-2 1 1 0 0 0 0 2z\\"></path>
          </svg>
          <div class=\\"reapop__notification-meta\\">
            <p class=\\"reapop__notification-message\\">hello world!</p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div class=\\"reapop__container reapop__container--bottom-right\\">
    <div>
      <div data-testid=\\"timed-notification\\">
        <div data-testid=\\"notification\\" role=\\"alert\\" class=\\"reapop__notification reapop__notification--info reapop__notification--buttons-0 \\"><svg viewBox=\\"0 0 16 16\\" fill=\\"currentColor\\" class=\\"reapop__notification-dismiss-icon\\">
            <path fill-rule=\\"evenodd\\" d=\\"M8 16A8 8 0 1 0 8 0a8 8 0 0 0 0 16zm.93-9.412l-2.29.287-.082.38.45.083c.294.07.352.176.288.469l-.738 3.468c-.194.897.105 1.319.808 1.319.545 0 1.178-.252 1.465-.598l.088-.416c-.2.176-.492.246-.686.246-.275 0-.375-.193-.304-.533L8.93 6.588zM8 5.5a1 1 0 1 0 0-2 1 1 0 0 0 0 2z\\"></path>
          </svg>
          <div class=\\"reapop__notification-meta\\">
            <p class=\\"reapop__notification-message\\">hello world!</p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>"
`;

exports[`<NotificationsSystem/> should display notifications in each container and then in the top container 2`] = `
"<div>
  <div class=\\"reapop__container reapop__container--top-center reapop__container--single\\">
    <div>
      <div data-testid=\\"timed-notification\\">
        <div data-testid=\\"notification\\" role=\\"alert\\" class=\\"reapop__notification reapop__notification--info reapop__notification--buttons-0 \\"><svg viewBox=\\"0 0 16 16\\" fill=\\"currentColor\\" class=\\"reapop__notification-dismiss-icon\\">
            <path fill-rule=\\"evenodd\\" d=\\"M8 16A8 8 0 1 0 8 0a8 8 0 0 0 0 16zm.93-9.412l-2.29.287-.082.38.45.083c.294.07.352.176.288.469l-.738 3.468c-.194.897.105 1.319.808 1.319.545 0 1.178-.252 1.465-.598l.088-.416c-.2.176-.492.246-.686.246-.275 0-.375-.193-.304-.533L8.93 6.588zM8 5.5a1 1 0 1 0 0-2 1 1 0 0 0 0 2z\\"></path>
          </svg>
          <div class=\\"reapop__notification-meta\\">
            <p class=\\"reapop__notification-message\\">hello world!</p>
          </div>
        </div>
      </div>
      <div data-testid=\\"timed-notification\\">
        <div data-testid=\\"notification\\" role=\\"alert\\" class=\\"reapop__notification reapop__notification--info reapop__notification--buttons-0 \\"><svg viewBox=\\"0 0 16 16\\" fill=\\"currentColor\\" class=\\"reapop__notification-dismiss-icon\\">
            <path fill-rule=\\"evenodd\\" d=\\"M8 16A8 8 0 1 0 8 0a8 8 0 0 0 0 16zm.93-9.412l-2.29.287-.082.38.45.083c.294.07.352.176.288.469l-.738 3.468c-.194.897.105 1.319.808 1.319.545 0 1.178-.252 1.465-.598l.088-.416c-.2.176-.492.246-.686.246-.275 0-.375-.193-.304-.533L8.93 6.588zM8 5.5a1 1 0 1 0 0-2 1 1 0 0 0 0 2z\\"></path>
          </svg>
          <div class=\\"reapop__notification-meta\\">
            <p class=\\"reapop__notification-message\\">hello world!</p>
          </div>
        </div>
      </div>
      <div data-testid=\\"timed-notification\\">
        <div data-testid=\\"notification\\" role=\\"alert\\" class=\\"reapop__notification reapop__notification--info reapop__notification--buttons-0 \\"><svg viewBox=\\"0 0 16 16\\" fill=\\"currentColor\\" class=\\"reapop__notification-dismiss-icon\\">
            <path fill-rule=\\"evenodd\\" d=\\"M8 16A8 8 0 1 0 8 0a8 8 0 0 0 0 16zm.93-9.412l-2.29.287-.082.38.45.083c.294.07.352.176.288.469l-.738 3.468c-.194.897.105 1.319.808 1.319.545 0 1.178-.252 1.465-.598l.088-.416c-.2.176-.492.246-.686.246-.275 0-.375-.193-.304-.533L8.93 6.588zM8 5.5a1 1 0 1 0 0-2 1 1 0 0 0 0 2z\\"></path>
          </svg>
          <div class=\\"reapop__notification-meta\\">
            <p class=\\"reapop__notification-message\\">hello world!</p>
          </div>
        </div>
      </div>
      <div data-testid=\\"timed-notification\\">
        <div data-testid=\\"notification\\" role=\\"alert\\" class=\\"reapop__notification reapop__notification--info reapop__notification--buttons-0 \\"><svg viewBox=\\"0 0 16 16\\" fill=\\"currentColor\\" class=\\"reapop__notification-dismiss-icon\\">
            <path fill-rule=\\"evenodd\\" d=\\"M8 16A8 8 0 1 0 8 0a8 8 0 0 0 0 16zm.93-9.412l-2.29.287-.082.38.45.083c.294.07.352.176.288.469l-.738 3.468c-.194.897.105 1.319.808 1.319.545 0 1.178-.252 1.465-.598l.088-.416c-.2.176-.492.246-.686.246-.275 0-.375-.193-.304-.533L8.93 6.588zM8 5.5a1 1 0 1 0 0-2 1 1 0 0 0 0 2z\\"></path>
          </svg>
          <div class=\\"reapop__notification-meta\\">
            <p class=\\"reapop__notification-message\\">hello world!</p>
          </div>
        </div>
      </div>
      <div data-testid=\\"timed-notification\\">
        <div data-testid=\\"notification\\" role=\\"alert\\" class=\\"reapop__notification reapop__notification--info reapop__notification--buttons-0 \\"><svg viewBox=\\"0 0 16 16\\" fill=\\"currentColor\\" class=\\"reapop__notification-dismiss-icon\\">
            <path fill-rule=\\"evenodd\\" d=\\"M8 16A8 8 0 1 0 8 0a8 8 0 0 0 0 16zm.93-9.412l-2.29.287-.082.38.45.083c.294.07.352.176.288.469l-.738 3.468c-.194.897.105 1.319.808 1.319.545 0 1.178-.252 1.465-.598l.088-.416c-.2.176-.492.246-.686.246-.275 0-.375-.193-.304-.533L8.93 6.588zM8 5.5a1 1 0 1 0 0-2 1 1 0 0 0 0 2z\\"></path>
          </svg>
          <div class=\\"reapop__notification-meta\\">
            <p class=\\"reapop__notification-message\\">hello world!</p>
          </div>
        </div>
      </div>
      <div data-testid=\\"timed-notification\\">
        <div data-testid=\\"notification\\" role=\\"alert\\" class=\\"reapop__notification reapop__notification--info reapop__notification--buttons-0 \\"><svg viewBox=\\"0 0 16 16\\" fill=\\"currentColor\\" class=\\"reapop__notification-dismiss-icon\\">
            <path fill-rule=\\"evenodd\\" d=\\"M8 16A8 8 0 1 0 8 0a8 8 0 0 0 0 16zm.93-9.412l-2.29.287-.082.38.45.083c.294.07.352.176.288.469l-.738 3.468c-.194.897.105 1.319.808 1.319.545 0 1.178-.252 1.465-.598l.088-.416c-.2.176-.492.246-.686.246-.275 0-.375-.193-.304-.533L8.93 6.588zM8 5.5a1 1 0 1 0 0-2 1 1 0 0 0 0 2z\\"></path>
          </svg>
          <div class=\\"reapop__notification-meta\\">
            <p class=\\"reapop__notification-message\\">hello world!</p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>"
`;

exports[`<NotificationsSystem/> should display notifications in each container and then in the top container 3`] = `
Array [
  Array [
    Object {
      "dismissNotification": [MockFunction],
      "notifications": Array [
        Object {
          "buttons": Array [],
          "id": "0",
          "message": "hello world!",
          "position": "top-center",
          "status": "info",
        },
      ],
      "position": "top-center",
      "singleContainer": false,
    },
    Object {},
  ],
  Array [
    Object {
      "dismissNotification": [MockFunction],
      "notifications": Array [
        Object {
          "buttons": Array [],
          "id": "1",
          "message": "hello world!",
          "position": "top-left",
          "status": "info",
        },
      ],
      "position": "top-left",
      "singleContainer": false,
    },
    Object {},
  ],
  Array [
    Object {
      "dismissNotification": [MockFunction],
      "notifications": Array [
        Object {
          "buttons": Array [],
          "id": "2",
          "message": "hello world!",
          "position": "top-right",
          "status": "info",
        },
      ],
      "position": "top-right",
      "singleContainer": false,
    },
    Object {},
  ],
  Array [
    Object {
      "dismissNotification": [MockFunction],
      "notifications": Array [
        Object {
          "buttons": Array [],
          "id": "3",
          "message": "hello world!",
          "position": "bottom-center",
          "status": "info",
        },
      ],
      "position": "bottom-center",
      "singleContainer": false,
    },
    Object {},
  ],
  Array [
    Object {
      "dismissNotification": [MockFunction],
      "notifications": Array [
        Object {
          "buttons": Array [],
          "id": "4",
          "message": "hello world!",
          "position": "bottom-left",
          "status": "info",
        },
      ],
      "position": "bottom-left",
      "singleContainer": false,
    },
    Object {},
  ],
  Array [
    Object {
      "dismissNotification": [MockFunction],
      "notifications": Array [
        Object {
          "buttons": Array [],
          "id": "5",
          "message": "hello world!",
          "position": "bottom-right",
          "status": "info",
        },
      ],
      "position": "bottom-right",
      "singleContainer": false,
    },
    Object {},
  ],
  Array [
    Object {
      "dismissNotification": [MockFunction],
      "notifications": Array [
        Object {
          "buttons": Array [],
          "id": "0",
          "message": "hello world!",
          "position": "top-center",
          "status": "info",
        },
        Object {
          "buttons": Array [],
          "id": "1",
          "message": "hello world!",
          "position": "top-left",
          "status": "info",
        },
        Object {
          "buttons": Array [],
          "id": "2",
          "message": "hello world!",
          "position": "top-right",
          "status": "info",
        },
        Object {
          "buttons": Array [],
          "id": "3",
          "message": "hello world!",
          "position": "bottom-center",
          "status": "info",
        },
        Object {
          "buttons": Array [],
          "id": "4",
          "message": "hello world!",
          "position": "bottom-left",
          "status": "info",
        },
        Object {
          "buttons": Array [],
          "id": "5",
          "message": "hello world!",
          "position": "bottom-right",
          "status": "info",
        },
      ],
      "position": "top-center",
      "singleContainer": true,
    },
    Object {},
  ],
]
`;

exports[`<NotificationsSystem/> should display notifications with custom components (transition and icon) 1`] = `
Array [
  Array [
    Object {
      "appear": undefined,
      "children": <NotificationContainer
        dismissNotification={[MockFunction]}
        notification={
          Object {
            "buttons": Array [],
            "id": "0",
            "message": "hello world!",
            "position": "top-center",
            "status": "info",
          }
        }
      >
        <Notification
          components={
            Object {
              "NotificationIcon": [Function],
              "Transition": [MockFunction] {
                "calls": [Circular],
                "results": Array [
                  Object {
                    "type": "return",
                    "value": <NotificationContainer
                      dismissNotification={[MockFunction]}
                      notification={
                        Object {
                          "buttons": Array [],
                          "id": "0",
                          "message": "hello world!",
                          "position": "top-center",
                          "status": "info",
                        }
                      }
                    >
                      <Notification
                        components={[Circular]}
                        dismissNotification={[MockFunction]}
                        notification={
                          Object {
                            "buttons": Array [],
                            "id": "0",
                            "message": "hello world!",
                            "position": "top-center",
                            "status": "info",
                          }
                        }
                      />
                    </NotificationContainer>,
                  },
                ],
              },
            }
          }
          dismissNotification={[MockFunction]}
          notification={
            Object {
              "buttons": Array [],
              "id": "0",
              "message": "hello world!",
              "position": "top-center",
              "status": "info",
            }
          }
        />
      </NotificationContainer>,
      "enter": undefined,
      "exit": undefined,
      "in": true,
      "notification": Object {
        "buttons": Array [],
        "id": "0",
        "message": "hello world!",
        "position": "top-center",
        "status": "info",
      },
      "onExited": [Function],
    },
    Object {},
  ],
]
`;

exports[`<NotificationsSystem/> should display notifications with custom components (transition and icon) 2`] = `
"<div>
  <div class=\\"reapop__container reapop__container--top-center\\">
    <div>
      <div data-testid=\\"timed-notification\\">
        <div data-testid=\\"notification\\" role=\\"alert\\" class=\\"reapop__notification reapop__notification--info reapop__notification--buttons-0 \\">
          <div>custom-icon</div>
          <div class=\\"reapop__notification-meta\\">
            <p class=\\"reapop__notification-message\\">hello world!</p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div class=\\"reapop__container reapop__container--top-left\\">
    <div></div>
  </div>
  <div class=\\"reapop__container reapop__container--top-right\\">
    <div></div>
  </div>
  <div class=\\"reapop__container reapop__container--bottom-center\\">
    <div></div>
  </div>
  <div class=\\"reapop__container reapop__container--bottom-left\\">
    <div></div>
  </div>
  <div class=\\"reapop__container reapop__container--bottom-right\\">
    <div></div>
  </div>
</div>"
`;

exports[`<NotificationsSystem/> should display notifications with custom components (transition and notification) 1`] = `
Array [
  Array [
    Object {
      "appear": undefined,
      "children": <NotificationContainer
        dismissNotification={[MockFunction]}
        notification={
          Object {
            "buttons": Array [],
            "id": "0",
            "message": "hello world!",
            "position": "top-center",
            "status": "info",
          }
        }
      >
        <CustomNotification
          components={
            Object {
              "Notification": [Function],
              "Transition": [MockFunction] {
                "calls": [Circular],
                "results": Array [
                  Object {
                    "type": "return",
                    "value": <NotificationContainer
                      dismissNotification={[MockFunction]}
                      notification={
                        Object {
                          "buttons": Array [],
                          "id": "0",
                          "message": "hello world!",
                          "position": "top-center",
                          "status": "info",
                        }
                      }
                    >
                      <CustomNotification
                        components={[Circular]}
                        dismissNotification={[MockFunction]}
                        notification={
                          Object {
                            "buttons": Array [],
                            "id": "0",
                            "message": "hello world!",
                            "position": "top-center",
                            "status": "info",
                          }
                        }
                      />
                    </NotificationContainer>,
                  },
                ],
              },
            }
          }
          dismissNotification={[MockFunction]}
          notification={
            Object {
              "buttons": Array [],
              "id": "0",
              "message": "hello world!",
              "position": "top-center",
              "status": "info",
            }
          }
        />
      </NotificationContainer>,
      "enter": undefined,
      "exit": undefined,
      "in": true,
      "notification": Object {
        "buttons": Array [],
        "id": "0",
        "message": "hello world!",
        "position": "top-center",
        "status": "info",
      },
      "onExited": [Function],
    },
    Object {},
  ],
]
`;

exports[`<NotificationsSystem/> should display notifications with custom components (transition and notification) 2`] = `
"<div>
  <div class=\\"reapop__container reapop__container--top-center\\">
    <div>
      <div data-testid=\\"timed-notification\\">
        <div>custom-notification</div>
      </div>
    </div>
  </div>
  <div class=\\"reapop__container reapop__container--top-left\\">
    <div></div>
  </div>
  <div class=\\"reapop__container reapop__container--top-right\\">
    <div></div>
  </div>
  <div class=\\"reapop__container reapop__container--bottom-center\\">
    <div></div>
  </div>
  <div class=\\"reapop__container reapop__container--bottom-left\\">
    <div></div>
  </div>
  <div class=\\"reapop__container reapop__container--bottom-right\\">
    <div></div>
  </div>
</div>"
`;

exports[`<NotificationsSystem/> should display notifications with default components 1`] = `
"<div>
  <div class=\\"reapop__container reapop__container--top-center\\">
    <div>
      <div data-testid=\\"timed-notification\\">
        <div data-testid=\\"notification\\" role=\\"alert\\" class=\\"reapop__notification reapop__notification--info reapop__notification--buttons-0 \\"><svg viewBox=\\"0 0 16 16\\" fill=\\"currentColor\\" class=\\"reapop__notification-dismiss-icon\\">
            <path fill-rule=\\"evenodd\\" d=\\"M8 16A8 8 0 1 0 8 0a8 8 0 0 0 0 16zm.93-9.412l-2.29.287-.082.38.45.083c.294.07.352.176.288.469l-.738 3.468c-.194.897.105 1.319.808 1.319.545 0 1.178-.252 1.465-.598l.088-.416c-.2.176-.492.246-.686.246-.275 0-.375-.193-.304-.533L8.93 6.588zM8 5.5a1 1 0 1 0 0-2 1 1 0 0 0 0 2z\\"></path>
          </svg>
          <div class=\\"reapop__notification-meta\\">
            <p class=\\"reapop__notification-message\\">hello world!</p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div class=\\"reapop__container reapop__container--top-left\\">
    <div></div>
  </div>
  <div class=\\"reapop__container reapop__container--top-right\\">
    <div></div>
  </div>
  <div class=\\"reapop__container reapop__container--bottom-center\\">
    <div></div>
  </div>
  <div class=\\"reapop__container reapop__container--bottom-left\\">
    <div></div>
  </div>
  <div class=\\"reapop__container reapop__container--bottom-right\\">
    <div></div>
  </div>
</div>"
`;

exports[`<NotificationsSystem/> should display notifications with theme 1`] = `
"<div>
  <div style=\\"position: fixed; z-index: 999999; top: 20px; left: 50%; transform: translateX(-50%);\\" class=\\"reapop__container reapop__container--top-center\\">
    <div>
      <div data-testid=\\"timed-notification\\">
        <div data-testid=\\"notification\\" role=\\"alert\\" style=\\"display: flex; width: 350px; height: 100%; position: relative; border-radius: 5px; border: 1px solid rgba(0,0,0,.1); box-shadow: 0 3px 8px 2px rgba(0, 0, 0, 0.3); background-color: rgb(47, 47, 54); color: rgb(255, 255, 255); margin-bottom: 20px; z-index: 999;\\" class=\\"reapop__notification reapop__notification--info reapop__notification--buttons-0 \\"><svg viewBox=\\"0 0 16 16\\" fill=\\"currentColor\\" style=\\"display: flex; width: 20px; height: 20px; box-sizing: border-box; margin: 10px 0px 10px 15px; align-self: flex-start; flex-shrink: 0; color: rgb(65, 146, 255);\\" class=\\"reapop__notification-dismiss-icon\\">
            <path fill-rule=\\"evenodd\\" d=\\"M8 16A8 8 0 1 0 8 0a8 8 0 0 0 0 16zm.93-9.412l-2.29.287-.082.38.45.083c.294.07.352.176.288.469l-.738 3.468c-.194.897.105 1.319.808 1.319.545 0 1.178-.252 1.465-.598l.088-.416c-.2.176-.492.246-.686.246-.275 0-.375-.193-.304-.533L8.93 6.588zM8 5.5a1 1 0 1 0 0-2 1 1 0 0 0 0 2z\\"></path>
          </svg>
          <div style=\\"vertical-align: top; box-sizing: border-box; width: 100%; padding: 10px 20px;\\" class=\\"reapop__notification-meta\\">
            <p style=\\"margin: 0px; font-size: 14px; color: rgb(185, 194, 203); line-height: 1.428571429;\\" class=\\"reapop__notification-message\\">hello world!</p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div style=\\"position: fixed; z-index: 999999; top: 20px; left: 20px;\\" class=\\"reapop__container reapop__container--top-left\\">
    <div></div>
  </div>
  <div style=\\"position: fixed; z-index: 999999; top: 20px; right: 20px;\\" class=\\"reapop__container reapop__container--top-right\\">
    <div></div>
  </div>
  <div style=\\"position: fixed; z-index: 999999; bottom: 20px; left: 50%; transform: translateX(-50%);\\" class=\\"reapop__container reapop__container--bottom-center\\">
    <div></div>
  </div>
  <div style=\\"position: fixed; z-index: 999999; bottom: 20px; left: 20px;\\" class=\\"reapop__container reapop__container--bottom-left\\">
    <div></div>
  </div>
  <div style=\\"position: fixed; z-index: 999999; bottom: 20px; right: 20px;\\" class=\\"reapop__container reapop__container--bottom-right\\">
    <div></div>
  </div>
</div>"
`;

exports[`<NotificationsSystem/> should display notifications without theme 1`] = `
"<div>
  <div class=\\"reapop__container reapop__container--top-center\\">
    <div>
      <div data-testid=\\"timed-notification\\">
        <div data-testid=\\"notification\\" role=\\"alert\\" class=\\"reapop__notification reapop__notification--info reapop__notification--buttons-0 \\"><svg viewBox=\\"0 0 16 16\\" fill=\\"currentColor\\" class=\\"reapop__notification-dismiss-icon\\">
            <path fill-rule=\\"evenodd\\" d=\\"M8 16A8 8 0 1 0 8 0a8 8 0 0 0 0 16zm.93-9.412l-2.29.287-.082.38.45.083c.294.07.352.176.288.469l-.738 3.468c-.194.897.105 1.319.808 1.319.545 0 1.178-.252 1.465-.598l.088-.416c-.2.176-.492.246-.686.246-.275 0-.375-.193-.304-.533L8.93 6.588zM8 5.5a1 1 0 1 0 0-2 1 1 0 0 0 0 2z\\"></path>
          </svg>
          <div class=\\"reapop__notification-meta\\">
            <p class=\\"reapop__notification-message\\">hello world!</p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div class=\\"reapop__container reapop__container--top-left\\">
    <div></div>
  </div>
  <div class=\\"reapop__container reapop__container--top-right\\">
    <div></div>
  </div>
  <div class=\\"reapop__container reapop__container--bottom-center\\">
    <div></div>
  </div>
  <div class=\\"reapop__container reapop__container--bottom-left\\">
    <div></div>
  </div>
  <div class=\\"reapop__container reapop__container--bottom-right\\">
    <div></div>
  </div>
</div>"
`;
